### PR TITLE
add functions for filtering

### PR DESCRIPTION
**Pull request by nicdnepr**
| Q | A |
| --- | --- |
| Is bugfix? | no |
| New feature? | yes |
| Breaks BC? | no |
| Tests pass? | no |
| Fixed issues | comma-separated list of tickets # fixed by the PR, if any |

### CODE REVIEW

File: framework/db/Query.php [hunk 1]
```diff
@@ -706,6 +706,24 @@ public function andHaving($condition, $params = [])
         return $this;
     }
 
+    public function andFilterHaving(array $condition)
```
Comment (samdark) [inline framework/db/Query.php]: Need `@since` and phpdoc.

File: framework/db/Query.php [hunk 2]
```diff
@@ -706,6 +706,24 @@ public function andHaving($condition, $params = [])
...
+    {
+        $condition = $this->filterCondition($condition);
+        if ($condition !== []) {
+            $this->andHaving($condition);
+        }
+        return $this;
+    }
+
+    public function orFilterHaving(array $condition)
```
Comment (samdark) [inline framework/db/Query.php]: Need `@since` and phpdoc.

**Comment by cebe** [comment]
@nicdnepr could you explain a use case for these methods?

**Comment by nicdnepr** [comment]
It's the same as function andFilterWhere, but only for having. I had a task to add GridView filtering for having results, and was surprised that there is no analog of function andFilterWhere. So I decided to add this code

**Review by samdark — CHANGES_REQUESTED** [review]
It looks good to me except `@since` and CHANGELOG.

**Review by SilverFire — CHANGES_REQUESTED** [review]
Tests are required

**Comment by samdark** [comment]
Not sure if it should be in QueryInterface...

**Comment by samdark** [comment]
> Not sure if it should be in QueryInterface...

No, it should not. It's in correct class.

**Comment by samdark** [comment]
@nicdnepr do you want to work on it further?

**Comment by nicdnepr** [comment]
@samdark Hi, I don't have much time at this moment

**Comment by samdark** [comment]
Closed in favor of https://github.com/yiisoft/yii2/pull/13210

**Pull request closed without merging**